### ISSUE TITLE
[Windows] Every update breaks taskbar pin

### ISSUE THREAD

**Issue by Enigma0** [plexinc/plex-media-player#425]
Every time PMP gets gets an update the installer breaks the taskbar pin and it reverts to a blank white document icon that can't find PMP when clicked (OS prompts for removal).

Been happening since around 1.1.2 IIRC.

**Comment by WayneD** [comment]
It seems to me like this could be a Windows bug. When PMP does its own update (and thus PMP starts running again), the now-broken icon is underlined, showing that Windows knows what executable it is associated with, but right-clicking the icon shows a bogus "pin to taskbar" choice instead of unpin. I usually do a pin + unpin + pin sequence that restores the icon to normal, and then I have to drag it back into position. Yeah, it's pretty annoying. I had hoped that the Plex folks would report this to MS by now, since I reported this issue in the forums ages ago, but nobody at Plex has ever responded in any way about this issue.

**Comment by Enigma0** [comment]
Yeah it's pretty irritating. Same problem with Mumble. 10 bucks says MS did this intentionally to sway folks into using more into their _cr_appstore.

That being said, the way PMP actually updates seems to be part of the problem - all other apps that receive updates and are pinned don't have this issue (besides mumble).

**Comment by Enigma0** [comment]
This is still a majorly irritating issue every update.

Every update breaks the pin and it has to be unpinned, re-opened, and re-pinned.

To make matters worse, I use PMP on an HTPC which has no mouse or keyboard and I have to remote in just to do this fix.

Meanwhile it also breaks the command arguments `--tv --fullscreen`.

**Comment by Enigma0** [comment]
Still broken in `1.3`

**Comment by LongChair** [comment]
@Enigma0 : just to make sure & update, is that still an issue with recent builds ?

**Comment by fvbommel** [comment]
@LongChair : I just updated to the latest version and had the blank white document icon again, and had to do the whole pin-unpin-pin-move back to correct position dance again (even though it was already pinned).
So yes, this is still an issue.

**Comment by LongChair** [comment]
Thanks for checking.

**Comment by Enigma0** [comment]
Yup still and issue for me as well.

**Comment by LongChair** [comment]
@markus101 : should we ping william about this ?

**Comment by LongChair** [comment]
could be related to https://github.com/plexinc/plex-media-player/issues/233

**Comment by LongChair** [comment]
@Enigma0 : do you have a setup that allows to build PMP installer ?
I have a few ideas that we should try, but i have a few others things on my plate at the moment.
If you can build it then i can provide a patch to test :)
Otherwise i'll look into that later

**Comment by LongChair** [comment]
@Enigma0 : could you please test the installer below and check that specific installer issue ?

https://www.dropbox.com/s/nezf5oztzy138c6/PlexMediaPlayer-2.10.0.840-5954ab87-windows-x64.exe?dl=0

**NOTE** Don't try to use the app itself as it's not released and might have still troubles.

It seems to fix the issue for me here, but would like to have a confirmation before i merge the fix.

**Comment by Enigma0** [comment]
@LongChair It still breaks with that installer also. I spun up a windows VM to test it.

**Comment by WayneD** [comment]
What does the nightly installer do differently from the regular installer? I have noticed before that it doesn't break the pinned taskbar icon.  I just confirmed this with the latest one: https://nightlies.plex.tv/public-test/plexmediaplayer/public-testing/2.9.0.831-b1acec41/PlexMediaPlayer-2.9.0.831-b1acec41-windows-x64.exe

**Comment by LongChair** [comment]
it doesn't remove shortcut on uninstall.
The installer will start with uninstalling previous install before updating to newer.
this process would break the shortcuts, and therefore taskbar pins.

**Comment by LongChair** [comment]
@Enigma0 : i'm very suprised that it still does, I tested this myself and got that tested by one of our ninjas. are you sure ?

**Comment by WayneD** [comment]
Yup, just re-tested going from 2.9.0.284 to 2.10.0.840 -- still busted.  Maybe the drop-box version cited above is not to the version with the installer change?

**Comment by LongChair** [comment]
@OttoKerner : isn't that the version i linked you ? i would think so.

**Comment by LongChair** [comment]
840 was the right one with that fix : https://github.com/plexinc/plex-media-player/commit/5954ab87d88b6d9fc5980fcbea13011524b47b15

**Comment by Enigma0** [comment]
Yup - the dropbox linked installer busted the taskbar shortcut

**Comment by LongChair** [comment]
yeah that wix installed is a nightmare :) But i'm holding on :)

**Comment by Enigma0** [comment]
Go Windows >.>

**Comment by WayneD** [comment]
Can't you just change the installer to be more like the nightly one? It doesn't trigger this bug.

**Comment by LongChair** [comment]
I'm not sure to understand which installer of which product you are referring to.
Did you check that creating the Pinned icon from Program Files exe works ?

**Comment by LongChair** [comment]
hmm that would be very off if it was the case, the build i linked comes from teh same nighly server.
The installer itself didn't get any change in over a year.
